packages/pretty-format/src/index.js

```diff
@@ -405,13 +456,13 @@ function prettyFormat(val, opts) {
     indent = createIndent(opts.indent);
     refs = [];
     const pluginsResult = printPlugin(val, indent, prevIndent, spacing, edgeSpacing, refs, opts.maxDepth, currentDepth, opts.plugins, opts.min, opts.callToJSON, opts.printFunctionName, opts.escapeRegex, colors);
-    if (pluginsResult) {
+    if (pluginsResult !== null) {
       return pluginsResult;
     }
   }
 
   const basicResult = printBasicValue(val, opts.printFunctionName, opts.escapeRegex);
-  if (basicResult) {
+  if (basicResult !== null) {
```

Comment:
On second thought, what do you think about:

* `if (basicResult !== null)` for the internal trusted code, because Flow will find `undefined` and so on
* `if (typeof plugin !== 'string')` for the external untrusted code, in case a serializer returns anything other than a string. Does that test cause a performance problem?